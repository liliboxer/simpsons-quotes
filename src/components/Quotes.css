.Quotes {
  list-style: none;
  display: flex;
  flex-wrap: wrap;
  width: 100%;

  img {
    width: 30%;
  }
}