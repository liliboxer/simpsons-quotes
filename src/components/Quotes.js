import React from 'react';
import PropTypes from 'prop-types';
import Quote from './Quote';

function Quotes({ quotes }) {
  const quoteList = quotes.map((quote, i) => (
    <li key={i}>
      <Quote 
        characterName={quote.characterName} 
        quote={quote.quote} 
        characterImage={quote.characterImage}
      />
    </li>
  ));

  return <ul>{quoteList}</ul>;
}

Quotes.propTypes = {
  quotes: PropTypes.arrayOf(PropTypes.shape({
    characterName: PropTypes.string.isRequired,
    quote: PropTypes.string.isRequired,
    characterImage: PropTypes.string.isRequired
  })).isRequired
};

export default Quotes;
