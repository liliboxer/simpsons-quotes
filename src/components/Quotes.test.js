import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('Quotes component', () => {
  it('renders Quotes', () => {
    const wrapper = shallow(<Quotes quotes={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
