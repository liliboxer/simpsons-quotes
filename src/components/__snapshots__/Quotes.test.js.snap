// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes component renders Quotes 1`] = `
<ul
  className="Quotes"
/>
`;
